internal/provider/slo_resource.go

```diff
@@ -0,0 +1,460 @@
+					stringvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("datasets")),
+				},
+			},
+			"datasets": schema.SetAttribute{
+				Description: "The datasets the SLO is evaluated on.",
+				Optional:    true,
+				Computed:    true,
+				ElementType: types.StringType,
+				PlanModifiers: []planmodifier.Set{
+					setplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.Set{
+					setvalidator.SizeAtLeast(1),
+					setvalidator.SizeAtMost(10),
+					setvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("dataset")),
+				},
+			},
+			"sli": schema.StringAttribute{
+				Description: `The alias of the Derived Column that will be used as the SLI to indicate event success.
+The derived column used as the SLI must be in the same dataset as the SLO. Additionally,
+the column evaluation should consistently return nil, true, or false, as these are the only valid values for an SLI.`,
+				Required: true,
+			},
+			"target_percentage": schema.Float64Attribute{
+				Description: "The percentage of qualified events that you expect to succeed during the `time_period`.",
+				Required:    true,
+				Validators: []validator.Float64{
+					float64validator.Between(0.00000, 99.9999),
+				},
+			},
+			"time_period": schema.Int64Attribute{
+				Description: "The time period, in days, over which your SLO will be evaluated.",
+				Required:    true,
+				Validators: []validator.Int64{
+					int64validator.AtLeast(1),
+				},
+			},
+			"tags": schema.MapAttribute{
+				Description: "A map of tags to assign to the resource.",
+				Optional:    true,
+				ElementType: types.StringType,
+				PlanModifiers: []planmodifier.Map{
+					modifiers.EquivalentTags(),
+				},
+				Validators: []validator.Map{
+					mapvalidator.SizeAtMost(client.MaxTagsPerResource),
+					mapvalidator.KeysAre(
+						stringvalidator.RegexMatches(client.TagKeyValidationRegex, "must only contain lowercase letters, and be 1-32 characters long"),
+					),
+					mapvalidator.ValueStringsAre(
+						stringvalidator.RegexMatches(client.TagValueValidationRegex, "must begin with a lowercase letter, be between 1-32 characters long, and only contain lowercase alphanumeric characters, -, or /"),
+					),
+				},
+			},
+		},
+	}
+}
+
+func (r *sloResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	dataset, id, found := strings.Cut(req.ID, "/")
+
+	// if dataset separator not found, we will assume its the bare id
+	// if thats the case, we need to reassign values since strings.Cut would return (id, "", false)
+	dsValue := types.StringNull()
+	idValue := types.StringValue(id)
+	if !found {
+		idValue = types.StringValue(dataset)
+	} else {
+		dsValue = types.StringValue(dataset)
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &models.SLOResourceModel{
+		ID:       idValue,
+		Dataset:  dsValue,
+		Datasets: types.SetNull(types.StringType),
+		Tags:     types.MapNull(types.StringType),
```

Comment:
nit: I think these should be `Unknown` as, well, we don't know :)
```suggestion
		Datasets: types.SetUnknown(types.StringType),
		Tags:     types.MapUnknown(types.StringType),
```